libs/vm_modules/tests/unit/fixed_point.cpp

```diff
@@ -28,21 +30,173 @@ class FixedPointTest : public ::testing::Test
     function main()
       print(1.0fp32);
     endfunction
   )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(1));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
 
-  ASSERT_TRUE(toolkit.Compile(TEXT));
-  ASSERT_TRUE(toolkit.Run());
+TEST_F(FixedPointTest, addition_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var a = 2.0fp32;
+      var b = 3.0fp32;
+      a += b;
+      print(a);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(5));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
 
-  double gt = static_cast<double>(fetch::fixed_point::fp32_t(1));
-  EXPECT_EQ(std::stod(stdout.str()), gt);
+TEST_F(FixedPointTest, subtraction_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var a = 3.0fp32;
+      var b = 2.0fp32;
+      a -= b;
+      print(a);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(1));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+
+TEST_F(FixedPointTest, multiplication_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var a = 3.0fp32;
+      var b = 2.0fp32;
+      a *= b;
+      print(a);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(6));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+
+TEST_F(FixedPointTest, divide_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var a = 3.0fp32;
+      var b = 2.0fp32;
+      a /= b;
+      print(a);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(1.5));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
 }
 
+TEST_F(FixedPointTest, array_32_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var myArray = Array<Fixed32>(5);
+
+      for (i in 0:4)
+        myArray[i] = toFixed32(i);
+      endfor
+      print(myArray[3]);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(3));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+
+TEST_F(FixedPointTest, array_64_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var myArray = Array<Fixed64>(5);
+
+      for (i in 0:4)
+        myArray[i] = toFixed64(i);
+      endfor
+      print(myArray[3]);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp64_t(3));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+//
+// TEST_F(FixedPointTest, map_32_fixed_point)
+//{
+//  char const *TEXT = R"(
+//    function main()
+//      var mymap = Map<Fixed32, Fixed32>();
+//      mymap[0] = 1;
+//      print(mymap[0]);
+//    endfunction
+//  )";
+//  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(1));
+//  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+//}
+
+TEST_F(FixedPointTest, sin_pi_32_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var pi : Fixed32 = 3.1415fp32;
```

Comment:
No need to specify the type when you define a variable using a literal: `var pi = 3.1415fp32;`. Ditto other tests here